common/lib/client/realtimechannel.js

```diff
@@ -110,11 +117,16 @@ var RealtimeChannel = (function() {
 			callback = flags;
 			flags = null;
 		}
-		callback = callback || function(err) {
-			if(err) {
-				Logger.logAction(Logger.LOG_MAJOR, 'RealtimeChannel.attach()', 'Channel attach failed: ' + err.toString());
+		if(!callback) {
+			if(Platform.promisify) {
```

Comment:
Doesn't this change the behaviour if `Platform.promisify` ? Ie there is no longer a log message?

Reply:
If the user does something with the returned promise (i.e. calling `.catch()`, they'll get the error there, so the situation's analogous to if they passed a callback.

If they're not -- i.e. if they're just doing `channel.attach()` with no callback, and throwing away the return value -- and they don't have an `unhandledRejection` handler, then then they'll get the error in an `UnhandledPromiseRejectionWarning`.

(...although trying now, they actually get `Unhandled promise rejection: [object Object]`, because it uses `util.toString` which is non-overridable. Which isn't great. But, well, they get _something_, so that should be a hint to them start handling the promise properly)

(If it caused the channel state to go to `failed` (rather than just `detached` or `suspended`), that will trigger a log message elsewhere.)